src/main/kotlin/org/rust/ide/core/overrideImplement/generateUtil.kt

```diff
@@ -0,0 +1,103 @@
+package org.rust.ide.core.overrideImplement
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.ui.SimpleColoredComponent
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.mixin.*
+import org.rust.lang.core.psi.util.trait
+import org.rust.utils.getFormattedParts
+import javax.swing.JTree
+
+sealed class TraitMember {
+    class Function(val function: RsFunction) : TraitMember()
+    class Type(val type: RsTypeAlias) : TraitMember()
+    class Constant(val constant: RsConstant) : TraitMember()
+}
```

Comment:
I just thought that if you use `RsNamedElement`, you can actually get rid of `TraitMember` and simplify things in this file. Functions, type aliases and constants are all `RsNamedElement`s, and hence inherit `name` and `getIcon()`, so there's no need in those `when`s in `renderTreeNode()` and `getText()`. The only place you need to know the exact type of the element, is when you define the `text`, but you can simply use `is RsFunction` etc there. And after my refactoring of #1020 we'll also remove this last `when`.